tiledb/sm/misc/types.h

```diff
@@ -168,6 +178,14 @@ class Range {
     return range_.empty() ? nullptr : &range_[end_pos];
   }
 
+  /** Copies 'end' into this range's end bytes for fixed-size ranges. */
+  void set_end(const void* const end) {
+    if (range_start_size_ != 0)
+      LOG_FATAL("Unexpected var-sized range.");
```

Comment:
Maybe add the something like `Cannot set end range` in the error message so that we know where this is thrown?